### PR TITLE
feat(search-service): adding /sitemap endpoint and new schema in sear…

### PR DESCRIPTION
**Pull request by ayushsatyam146**
# Closes/Fixes/Resolves
ONEPLAT-2405
### 

# Explain the feature/fix

Adding a new `/sitemap` REST endpoint in the search-service along with the new Schema to store the web crawling configuration for static web apps.

## Does this PR introduce a breaking change

No

### CODE REVIEW

File: packages/search-service/src/search-sitemap/schema.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {
+  Document, Model, model, Schema,
+} from 'mongoose';
+
+export const SiteMapSchema: Schema = new Schema({
+  url: { type: String },
+  contentType: { type: String },
```
Comment (riginoommen) [inline packages/search-service/src/search-sitemap/schema.ts:7]: i think lastModified to be in the db

Comment (deshmukhmayur) [reply packages/search-service/src/search-sitemap/schema.ts:7]: Last modified is not needed, as that will be a inferred field and thus should not be stored in the collection.

File: packages/search-service/src/search-sitemap/resolver.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Request } from 'express';
+import { SiteMaps } from './schema';
+
+export const getSiteMap = async (req: Request) => {
+  const { contentType } = req.query;
+
+  const SiteMapdata = await (
```
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/resolver.ts:7]: Use small camel case for variables. Reserve capital for Classes, Interfaces or Types.
```suggestion
  const sitemapData = await (
```

File: packages/search-service/src/search-sitemap/types.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+declare type SiteMap = {
```
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/types.d.ts:1]:
```suggestion
declare type Sitemap = {
```

File: packages/search-service/src/search-sitemap/resolver.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    await SiteMaps.find()
+  ).map((item) => ({
+    url: item.url,
+    contentType: item.contentType,
+    lastModified: new Date().toISOString(),
+  }));
+
+  const FiltereSiteMapdata = contentType
```
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/resolver.ts:15]:
```suggestion
  const filteredSitemapData = contentType
```

File: packages/search-service/src/search-sitemap/resolver.ts (hunk 2 above)
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/resolver.ts:7]: `await` is not needed here as the return type of the `.map` is not a promise. Only 1 `await` is enough for the `Sitemaps.find`
```suggestion
  const SiteMapdata = (
```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/search-service/src/search-sitemap/resolver.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    lastModified: new Date().toISOString(),
+  }));
+
+  const filteredSitemapData = contentType
+    ? sitemapData.filter((item) => item.contentType === contentType)
+    : sitemapData;
+
+  return {
+    count: filteredSitemapData.length,
+    filteredSitemapData,
```
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/resolver.ts:19]: Missed this in my first review
```suggestion
    data: filteredSitemapData,
```

File: packages/search-service/src/search-sitemap/schema.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+});
+
+interface SiteMapModel extends Sitemap, Document {}
+
+interface SiteMapStatic extends Model<SiteMapModel> {}
+
+export const SiteMaps: Model<SiteMapModel> = model<
+SiteMapModel,
+SiteMapStatic
+>('SiteMap', SiteMapSchema);
```
Comment (deshmukhmayur) [inline packages/search-service/src/search-sitemap/schema.ts:17]: Can you format this properly?

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

**Review by deshmukhmayur — APPROVED** [review]

**Review by riginoommen — APPROVED** [review]

**Review by riginoommen — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**